Feat: script editor in `pre` and `post` backup scripts

SAMAD101 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

### Description
<!--- Describe your changes in detail -->
I've added buttons beside the `pre-backup` and `post-backup` line edits in the `Schedule` tab in the `Shell Commands` section.

https://github.com/borgbase/vorta/assets/71956678/71676ac1-5db2-4a34-8adc-f7f3b9033907

![image](https://github.com/borgbase/vorta/assets/71956678/975814e7-c224-4376-a7d1-d7bad26f37b1)


### Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
#1945 

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Makes it more convenient to add the commands using a shell script.

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
does effect much of the code, only extends some functionality

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the [CONTRIBUTING](https://vorta.borgbase.com/contributing/) guide.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.


*I provide my contribution under the terms of the [license](./../LICENSE.txt) of this repository and I affirm the [Developer Certificate of Origin][dco].*

[dco]: https://developercertificate.org/

<!--
This template is sourced from the awesome https://github.com/TalAter/open-source-templates
-->

SAMAD101 committed:
added peewee to requirements.d/dev.txt

SAMAD101 committed:
removed peewee from requirements: not necessary

SAMAD101 committed:
Merge branch 'borgbase:master' into master

SAMAD101 committed:
Merge branch 'borgbase:master' into master

SAMAD101 committed:
Merge branch 'borgbase:master' into master

SAMAD101 committed:
Merge branch 'borgbase:master' into master

SAMAD101 committed:
Merge branch 'borgbase:master' into master

SAMAD101 committed:
Merge branch 'borgbase:master' into master

SAMAD101 committed:
Merge branch 'borgbase:master' into master

SAMAD101 committed:
Merge branch 'borgbase:master' into master

SAMAD101 committed:
Merge branch 'borgbase:master' into master

SAMAD101 committed:
Merge branch 'borgbase:master' into master

SAMAD101 committed:
Merge branch 'borgbase:master' into master

SAMAD101 committed:
feat: file selection dialog for selecting pre and post backup script in schedule tab

shivansh02 commented:
Oh even I was ready with a PR, I shared a screenshot in the issue thread. Could've saved one of us the work if you told me that you were working on this :sweat_smile:

SAMAD101 commented:
> Oh even I was ready with a PR, I shared a screenshot in the issue thread. Could've saved one of us the work if you told me that you were working on this 😅

damn 😔 yeah I should've. I just finished this one today.

m3nu commented:
To avoid overlaps, you can make sure the issue is assigned to you before working on it.

As for this implementation, it's reading a file. We should also look into having a larger editor window instead. Often it's hard to decide what's the best way without seeing it. So even PRs that aren't merged can help with discovering a good solution.

SAMAD101 commented:
> To avoid overlaps, you can make sure the issue is assigned to you before working on it.
> 
> As for this implementation, it's reading a file. We should also look into having a larger editor window instead. Often it's hard to decide what's the best way without seeing it. So even PRs that aren't merged can help with discovering a good solution.

Thank you 
Will try implementing that as well.

PS: please assign me the issue #1945

SAMAD101 commented:
Well, apparently I have exams starting soon and also have to ready my proposal for GSOC. So, I'll add this one along with some more issues in my proposal.

SAMAD101 committed:
feat: editor for pre and post backup cmd

SAMAD101 commented:
> To avoid overlaps, you can make sure the issue is assigned to you before working on it.
> 
> As for this implementation, it's reading a file. We should also look into having a larger editor window instead. Often it's hard to decide what's the best way without seeing it. So even PRs that aren't merged can help with discovering a good solution.

Well, I snatched some time and implemented a small editor window for it. I've updated this PR. This is how it looks so far:
![Screenshot_20240303_012208](https://github.com/borgbase/vorta/assets/71956678/07c33083-7764-4b8f-9e45-b0d81588d62d)

m3nu commented:
Editor looks good. Just what I imagined.

I'd remove the "Load from file" button though. Since the problem it solves is very minor. One can easily copy and paste in the editor. Would just make sure the editor window can be resized.

SAMAD101 commented:
> Editor looks good. Just what I imagined.
> 
> I'd remove the "Load from file" button though. Since the problem it solves is very minor. One can easily copy and paste in the editor. Would just make sure the editor window can be resized.

sure, that can be done.  Yeah, The editor window is resizable, will make it wider ig.

SAMAD101 commented:
Update:
![Screenshot_20240304_135711](https://github.com/borgbase/vorta/assets/71956678/b486d290-1c48-4e0b-bd78-da42b57f3279)
![Screenshot_20240304_135717](https://github.com/borgbase/vorta/assets/71956678/939f05df-888b-4a5a-b56d-39e6a5dcd82a)

SAMAD101 committed:
Not needed: Removed the add from file button/dialog.

shivansh02 commented:
some spacing between the field and Save button would look good IMO.

m3nu commented:
Looks good to me. Will try it locally and then merge, if there are no other concerns or things that come up.

SAMAD101 committed:
minor changes: spacing between textedit and save button

SAMAD101 commented:
**Updated**

| Before | ![Screenshot_20240304_135711](https://github.com/borgbase/vorta/assets/71956678/0e451052-2264-4d5a-929f-636d49de36b0)  | 
|-|-|
|After | ![Screenshot_20240304_152540](https://github.com/borgbase/vorta/assets/71956678/ba65c2ca-61ae-4fe3-822d-72fc8a3dc3f6) |

diivi reviewed (commented):
IMO the design looks a bit off, attaching some issues:

![Too much space here](https://github.com/borgbase/vorta/assets/41837037/de570baf-0baa-41e1-a7e9-c7e03456fa55)

I did something similar with the Exclude GUI last year, the UI for that looked decent and referring to that might be good for consistency. Here's a screenshot:

![image](https://github.com/borgbase/vorta/assets/41837037/8cd1736e-9923-437e-ae2e-30949b9175f4)

(Ignore the tabbed button above) Nice work otherwise!

Also, the title ("Edit Pre-Backup Script") for the dialogs is different but the header text ("Custom Pre/Post backup script") is still the same, maybe you can change that too?

SAMAD101 committed:
minor changes: script editor dialog

SAMAD101 commented:
**Updated:**
- Now the label does change depending upon the context (pre/post-backup script).
- Controlled the spacing. Looks more aligned now.

![Screenshot_20240305_134224](https://github.com/borgbase/vorta/assets/71956678/8a1724d5-a16b-4549-a339-679bcfe330ba)

SAMAD101 commented:
Now that it's complete, how about I add some syntax highlighting to it as well? Need some thoughts.

real-yfprojects commented:
> Now that it's complete, how about I add some syntax highlighting to it as well? Need some thoughts.

Seems to much efford for such a small use case. Most people will copy paste the script from another editor anyways.

SAMAD101 commented:
> Seems to much efford for such a small use case. Most people will copy paste the script from another editor anyways.

Yeah, I'll drop the idea. Its fine as it is.

SAMAD101 committed:
Merge branch 'master' into issue1945

SAMAD101 committed:
Merge branch 'borgbase:master' into issue1945
